### DISCUSSION TITLE
How to handle null IEnumerable (Misconfiguration)

### DISCUSSION THREAD

**Discussion by cristianUrbina** [riok/mapperly#552] [Q&A]
Hello!! thanks for taking the time to check this out.

This is my mapper: 
```
    [Mapper(
        EnumMappingStrategy = EnumMappingStrategy.ByName,
        UseReferenceHandling = true,
        ThrowOnMappingNullMismatch = false,
        ThrowOnPropertyMappingNullMismatch = false
        )]
    public static partial class FirstTestDtoMapper
    {
        [MapProperty(nameof(FirstTest.Age), nameof(FirstTestDto.CurrentAge))]
        public static partial FirstTestDto FirstTestToDto(FirstTest firstTest);
        public static partial PersonDto PersonToDto(Person person);
    }
```

And this is the test that is not passing because a null IEnumerable (the children list):

![image](https://github.com/riok/mapperly/assets/57534097/153bc775-b0cd-467f-af98-3bae7984b681)
```
[Fact]
        public void FirstTestDtoMapper_PlainPerson_ShouldBeEqual()
        {
            // arrange
            var person = new Person()
            {
                SSN = 4,
                Name = "Josefina",
            };

            // act
            PersonDto dto = FirstTestDtoMapper.PersonToDto(person);

            // assert
            Assert.NotNull(dto);
            Assert.Equal(person.SSN, dto.SSN);
            Assert.Equal(person.Name, dto.Name);
        }
```

```
public class Person
    {
        public int SSN { get; set; }
        public string Name { get; set; }
        public IEnumerable<Person> Children { get; set; }
        public Person Father { get; set; }
        public Person Mother { get; set; }
        public Person FavoritePerson { get; set; }
    }
```
The workaround I can think of is performing some checking and cleaning using this https://mapperly.riok.app/docs/configuration/before-after-map. Is there a way to handle this without adding any Before / after map code?
I have already tried using Mapper attributes (those you see in the mapper class code).
Thanks again

**Comment by TimothyMakkison** [comment]
Hey, thanks for using Mapperly. 

Do you have nullable reference types enabled?
Looking at the generated code it looks like mapperly doesn't expect the value `Children` to be nullable (because nullable reference types is enabled). Because of this it doesn't add a null check to `Children`,resulting in `Select` throwing an error.

I think you can resolve this issue by annotating your properties like so:
```C#
public class Person
{
        public int SSN { get; set; }
        public string Name { get; set; }
        public IEnumerable<Person?>? Children { get; set; }
        public Person? Father { get; set; }
        public Person? Mother { get; set; }
        public Person? FavoritePerson { get; set; }
}
```

I wouldn't recommend this but adding `#nullable disable` before `Person` and `PersonDto` will disable nullable reference types, letting mapperly treat each property as a potential null value.
